Fix JPEG support code compilation on Android & Emscripten

peter-b opened the pull request:
The Android and Emscripten builds are unhappy with the updated
libjpeg, which appears to define its `boolean` type and `FALSE`/`TRUE`
constants slightly differently.

Update: This appears to also fix a build failure on iOS 10.

peter-b commented:
At the moment I expect this PR to succeed on Android, but fail on Emscripten and Windows builders. Still trying to work out what the [extremely funky Emscripten compiler tantrum](https://vulcan.livecode.com/builders/try-community-emscripten/builds/504/steps/compile/logs/stdio) is all about.

livecodeian commented:
@livecode-vulcan review ok e33bbed

livecode-vulcan commented:
💙 review by @livecodeian ok e33bbed542a42caa8108d4cee81e0da1ef2c2779

livecode-vulcan commented:
😞 test failure e33bbed542a42caa8108d4cee81e0da1ef2c2779
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/510)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/515)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/510)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/514)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/517)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/510)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/511)

peter-b committed:
Fix JPEG support code compilation on Android & Emscripten

The Android and Emscripten builds are unhappy with the updated
libjpeg, which appears to define its `boolean` type and `FALSE`/`TRUE`
constants slightly differently.

peter-b commented:
Rebased to `develop-8.1` because this patch is also needed there.

@livecode-vulcan review by @livecodeian ok 3a71bf2

livecode-vulcan commented:
💙 review by @livecodeian ok 3a71bf2e688847593973d7bdcdda8a64df69926c

livecode-vulcan commented:
😞 test failure 3a71bf2e688847593973d7bdcdda8a64df69926c
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/511)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/516)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/511)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/515)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/518)
- try-community-win-x86: [failure](https://vulcan.livecode.com/builders/try-community-win-x86/builds/511)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/512)

peter-b commented:
I'm going to merge this immediately since it resolves the Emscripten, iOS and Android build failures.
